src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DirectiveAttributeCompletionContext.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Razor.Completion;
+
+internal record DirectiveAttributeCompletionContext(
+    string SelectedAttributeName = "",
+    string? SelectedParameterName = null,
+    ImmutableArray<string> ExistingAttributes = default,
```

Comment:
Consider changing this type so `ExistingAttributes` is `[]` instead of default when not set. Adding a little more code here will help avoid extra `IsDefault` checks below.